lib/msal-node/src/client/ManagedIdentityClient.ts

```diff
@@ -32,6 +33,7 @@ export class ManagedIdentityClient {
     private nodeStorage: NodeStorage;
     private networkClient: INetworkModule;
     private cryptoProvider: CryptoProvider;
+    private retryTracker: RetryTracker;
```

Comment:
I don't think the lifetime of this object is correct. There should be 1 retry possible for each HTTP call. I think here you will have just 1 try per ManagedIdentityClient, not per request.

Reply:
Correct, but this is where the resetNumRetried() method comes into play. I reset the counter of retriedRequests to 0 after each request (retried request, if applicable).